File: invesdwin-scripting-parent/invesdwin-scripting-julia/invesdwin-scripting-julia-runtime-contract/src/test/java/de/invesdwin/scripting/julia/runtime/contract/InputsAndResultsTestByte.jl
println("getByte")
if isdefined(Main, :getByte) && !isnothing(getByte)
	error("getByte already defined!")
end
getByte = putByte
println(typeof(getByte))
println(getByte)
if typeof(getByte) != Int8
	error("getByte not Int8!")
end

println("getByteVector")
if isdefined(Main, :getByteVector) && !isnothing(getByteVector)
	error("getByteVector already defined!")
end
getByteVector = putByteVector
println(eltype(getByteVector))
println(getByteVector)
if eltype(getByteVector) != Int8
	error("getByteVector not Int8!")
end

println("getByteVectorAsList")
if isdefined(Main, :getByteVectorAsList) && !isnothing(getByteVectorAsList)
	error("getByteVectorAsList already defined!")
end
getByteVectorAsList = putByteVectorAsList
println(eltype(getByteVectorAsList))
println(getByteVectorAsList)
if eltype(getByteVectorAsList) != Int8
	error("getByteVectorAsList not Int8!")
end

println("getByteMatrix")
if isdefined(Main, :getByteMatrix) && !isnothing(getByteMatrix)
	error("getByteMatrix already defined!")
end
getByteMatrix = putByteMatrix
println(eltype(getByteMatrix))
println(getByteMatrix)
if eltype(getByteMatrix) != Int8
	error("getByteMatrix not Int8!")
end

println("getByteMatrixAsList")
if isdefined(Main, :getByteMatrixAsList) && !isnothing(getByteMatrixAsList)
	error("getByteMatrixAsList already defined!")
end
getByteMatrixAsList = putByteMatrixAsList
println(eltype(getByteMatrixAsList))
println(getByteMatrixAsList)
if eltype(getByteMatrixAsList) != Int8
	error("getByteMatrixAsList not Int8!")
end

File: invesdwin-scripting-parent/invesdwin-scripting-runtime-lua/src/test/java/de/invesdwin/scripting/runtime/lua/tests/InputsAndResultsTestByte.lua
arraysToString = java.method(java.import('de.invesdwin.util.collections.Arrays'),'toString','java.lang.Object')

print("getByte")
if(type(getByte) ~= "nil") then
	error("getByte already defined!")
end
getByte = putByte
getByteType = type(getByte)
print(getByteType)
print(getByte)
if(getByteType ~= "number") then
	error("getByte not Byte!")
end

print("getByteVector")
if(type(getByteVector) ~= "nil") then
	error("getByteVector already defined!")
end
getByteVector = putByteVector
getByteVectorType = type(getByteVector[1])
print(getByteVectorType)
print(arraysToString(getByteVector))
if(getByteVectorType ~= "number") then
	error("getByteVector not Byte!")
end

print("getByteVectorAsList")
if(type(getByteVectorAsList) ~= "nil") then
	error("getByteVectorAsList already defined!")
end
getByteVectorAsList = putByteVectorAsList
getByteVectorAsListType = type(getByteVectorAsList[1])
print(getByteVectorAsListType)
print(arraysToString(getByteVectorAsList))
if(getByteVectorAsListType ~= "number") then
	error("getByteVectorAsList not Byte!")
end

print("getByteMatrix")
if(type(getByteMatrix) ~= "nil") then
	error("getByteMatrix already defined!")
end
getByteMatrix = putByteMatrix
getByteMatrixType = type(getByteMatrix[1][1])
print(getByteMatrixType)
print(arraysToString(getByteMatrix))
if(getByteMatrixType ~= "number") then
	error("getByteMatrix not Byte!")
end

print("getByteMatrixAsList")
if(type(getByteMatrixAsList) ~= "nil") then
	error("getByteMatrixAsList already defined!")
end
getByteMatrixAsList = putByteMatrixAsList
getByteMatrixAsListType = type(getByteMatrixAsList[1][1])
print(getByteMatrixAsListType)
print(arraysToString(getByteMatrixAsList))
if(getByteMatrixAsListType ~= "number") then
	error("getByteMatrixAsList not Byte!")
end
